### PR TITLE
Change error level for `WordPress.WhiteSpace.CastStructureSpacing` checks to `error`.

### LINKED ISSUE #686: Change cast spacing warnings to errors
See https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/648#discussion_r78302016.

The cast spacing is called for specifically by the coding standards, so it should be an error like every other such violation, not just a warning.

### PR DESCRIPTION
**Pull request by jrfnl**
Fixes #686

### CODE REVIEW

**Comment by JDGrimes** [comment]
@westonruter @GaryJones if you have no objections to this, feel free to merge.

File: WordPress-Core/ruleset.xml [hunk 1]
```diff
@@ -150,9 +150,7 @@
 		<!-- Covers rule: When type casting, do it like so: $foo = (boolean) $bar; -->
 		<rule ref="Generic.Formatting.SpaceAfterCast"/>
 		<rule ref="WordPress.WhiteSpace.CastStructureSpacing"/>
-		<rule ref="WordPress.WhiteSpace.CastStructureSpacing.ContainsWhiteSpace">
-			<type>warning</type>
-		</rule>
+		<rule ref="WordPress.WhiteSpace.CastStructureSpacing.ContainsWhiteSpace"/>
```
Comment (GaryJones) [inline WordPress-Core/ruleset.xml]: Line 152 above already includes the whole of CastStructureSpacing, so is this extra explicit line 153 really needed? I think probably not, now where not passing it any custom config.

**Comment by GaryJones** [comment]
No objections, but I think the code can be tweaked (see comment).

File: WordPress-Core/ruleset.xml (hunk 1 above)
Comment (jrfnl) [inline WordPress-Core/ruleset.xml]: Good catch! Adjusted.

**Pull request merged**